# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmValue.h
# Commit: 72bf9c871ded
# Full Hash: 72bf9c871ded006f56f09b24d0723365516445bb
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2023-08-05 09:19:01
# Regressor Bug: 1692065
# File Overlap Count: 5
# Description:
#   Bug 1692065 - wasm: Add tagging for 31-bit integers and i31ref. r=yury
#   
#   This commit adds a tag 0x1 for 31-bit integers in AnyRef. Any JS value
#   that is losslessly representable (without coercion) in this range is
#   stored without boxing. This is chosen to line up exactly with how the
# ==============================================================================

diff -r 28feda14fdec -r 72bf9c871ded js/src/wasm/WasmValue.h
--- a/js/src/wasm/WasmValue.h	Fri Aug 04 14:06:50 2023 +0000
+++ b/js/src/wasm/WasmValue.h	Fri Aug 04 14:06:50 2023 +0000
@@ -380,6 +380,10 @@
 [[nodiscard]] extern bool CheckEqRefValue(JSContext* cx, HandleValue v,
                                           MutableHandleAnyRef vp);
 
+// The same as above for when the target type is 'i31ref'.
+[[nodiscard]] extern bool CheckI31RefValue(JSContext* cx, HandleValue v,
+                                           MutableHandleAnyRef vp);
+
 // The same as above for when the target type is 'structref'.
 [[nodiscard]] extern bool CheckStructRefValue(JSContext* cx, HandleValue v,
                                               MutableHandleAnyRef vp);
